src/GeniusYield/Test/Utils.hs

```diff
@@ -236,6 +225,30 @@ withWalletBalancesCheck ((w, v) : xs) m = do
         fail $ printf "expected balance difference of %s for wallet %s, but the actual difference was %s" v (walletName w) diff
     return b
 
+{- | Computes a 'GYTxMonadRun' action, checking that the 'Wallet' balances
+        change according to the input list. This is a simplified version of `withWalletBalancesCheck` where the input list need not consider lovelaces required for fees & to satisfy the min ada requirements as these are added automatically. It is therefore recommended to use this function over `withWalletBalancesCheck` to avoid hardcoding the lovelaces required for fees & min ada constraints.
+
+Notes:
+* An empty list means no checks are performed.
+* The 'GYValue' should be negative to check if the Wallet lost those funds.
+-}
+withWalletBalancesCheckSimple :: [(Wallet, GYValue)] -> GYTxMonadRun a -> GYTxMonadRun a
```

Comment:
Will it be nice to have the "equivalent" function but with signature `[(Wallet, GYValue)] -> Run a -> Run a`?

Reply:
Changes exist in our type `GYTxMonadRun`, so I don't think it's possible for function with such a signature.

Reply:
Right, I thought maybe it's a way. I will keep thinking :) Either way, this is ready to go. Thanks :)